mlx_graphs/datasets/ogb_dataset.py

```diff
@@ -0,0 +1,194 @@
+
+    def process(self):
+        try:
+            dataset = self._raw_ogb_dataset
+        except AttributeError:
+            # reload if already downloaded
+            self._load_data()
+            dataset = self._raw_ogb_dataset
+
+        if self.name in get_args(OGB_NODE_DATASET):
+            graph, label = dataset[0]
+            split_idx: dict = dataset.get_idx_split()  # type: ignore
+            train_idx, val_idx, test_idx = (
+                split_idx["train"],
+                split_idx["valid"],
+                split_idx["test"],
+            )
+            num_nodes = graph["num_nodes"]
+            self.graphs.append(
+                GraphData(
+                    edge_index=to_mx_array(graph["edge_index"]),
+                    node_features=to_mx_array(graph["node_feat"]),
+                    edge_features=to_mx_array(graph["edge_feat"]),
+                    node_labels=to_mx_array(label),  # type: ignore
+                    train_mask=index_to_mask(to_mx_array(train_idx), size=num_nodes),
+                    val_mask=index_to_mask(to_mx_array(val_idx), size=num_nodes),
+                    test_mask=index_to_mask(to_mx_array(test_idx), size=num_nodes),
+                )
+            )
+        elif self.name in get_args(OGB_EDGE_DATASET):
+            graph = dataset[0]
+            split_idx: dict = dataset.get_edge_split()  # type: ignore
+            train_edges, val_edges, test_edges = (
+                split_idx["train"]["edge"].reshape(2, -1),
+                split_idx["valid"]["edge"].reshape(2, -1),
+                split_idx["test"]["edge"].reshape(2, -1),
+            )
+            self.graphs.append(
+                GraphData(
+                    edge_index=to_mx_array(graph["edge_index"]),
+                    node_features=to_mx_array(graph["node_feat"]),
+                    edge_features=to_mx_array(graph["edge_feat"]),
+                    train_edge_index=to_mx_array(train_edges),
+                    val_edge_index=to_mx_array(val_edges),
+                    test_edge_index=to_mx_array(test_edges),
+                )
+            )
+
+        elif self.name in get_args(OGB_GRAPH_DATASET):
+            for graph, label in dataset:
+                self.graphs.append(
+                    GraphData(
+                        edge_index=to_mx_array(graph["edge_index"]),
+                        node_features=to_mx_array(graph["node_feat"]),
+                        edge_features=to_mx_array(graph["edge_feat"]),
+                        node_labels=to_mx_array(label),  # type: ignore
+                    )
+                )
+            split_idx: dict = dataset.get_idx_split()  # type: ignore
+            train_idx, val_idx, test_idx = (
+                split_idx["train"],
+                split_idx["valid"],
+                split_idx["test"],
+            )
+            if self.split == "train":
+                self.graphs = [self.graphs[i] for i in train_idx]
```

Comment:
did you try `self.graphs = self.graphs[train_idx]`?

Reply:
yes, it doesn't work, even if `train_idx` is converted to a list

Reply:
hmm, strange